node/src/components/linear_chain_sync/operations.rs

```diff
@@ -484,19 +522,170 @@ pub(crate) async fn run_fast_sync_task(
     }
 
     // Synchronize the trie store for the most recent block header.
-    info!(
-        state_root_hash = ?most_recent_block_header.state_root_hash(),
-        "Syncing trie store",
+    sync_trie_store(effect_builder, *most_recent_block_header.state_root_hash()).await?;
+
+    Ok((trusted_key_block_info, most_recent_block_header))
+}
+
+/// Downloads and saves the deploys and transfers for a block.
+async fn sync_deploys_and_transfers_and_state(
+    effect_builder: EffectBuilder<JoinerEvent>,
+    block: &Block,
+) -> Result<(), LinearChainSyncError> {
+    for deploy_hash in block.deploy_hashes() {
+        fetch_and_store_deploy(effect_builder, *deploy_hash).await?;
+    }
+    for transfer_hash in block.transfer_hashes() {
+        fetch_and_store_deploy(effect_builder, *transfer_hash).await?;
+    }
```

Comment:
This code will make 2500 (+ 100 for Wasm) sequential requests to random peers, correct? This will be *very* slow.